java/amazon-kinesis-producer/src/main/java/com/amazonaws/services/kinesis/producer/KinesisProducer.java

```diff
@@ -132,7 +165,13 @@ public void run() {
                     } else if (m.hasMetricsResponse()) {
                         onMetricsResponse(m);
                     } else {
-                        log.error("Unexpected message type from child process");
+                        // clear the future here as well since the native core has exhausted its retries.
+                        SettableFuture<UserRecordResult> f = getFuture(m);
+                        f.setException(new UnexpectedMessageException("Unexpected message type from child process"));
+                        log.error(String.format("Unexpected message type with case %s from child process with message"
+                                + " id %s. Removing the submitted future from processing queue.",
+                                m.getActualMessageCase().toString(), m.getSourceId()));
```

Comment:
Can m.getActualMessageCase be null? If yes, we need to handle null check.

Reply:
removed toString() since ActualMessageCase is an enum